rest/handler.go

```diff
@@ -648,7 +664,10 @@ func (h *handler) checkAuth(dbCtx *db.DatabaseContext) (err error) {
 		if dbCtx.Options.SendWWWAuthenticateHeader == nil || *dbCtx.Options.SendWWWAuthenticateHeader {
 			h.response.Header().Set("WWW-Authenticate", wwwAuthenticateHeader)
 		}
-		return base.HTTPErrorf(http.StatusUnauthorized, "Login required")
+		if h.providedAuthCredentionals() {
```

Comment:
This is hit by 

```
        response = rt.SendUserRequest(http.MethodGet, "/{{.db}}/", "", "user1")
	RequireStatus(t, response, http.StatusUnauthorized)
	require.Contains(t, response.Body.String(), ErrInvalidLogin.Message)
```

from `TestDisablePublicBasicAuth`

Reply:
I thought it would be preferable to do this check earlier in checkAuth (i.e. if DisablePasswordAuth is true, return ErrInvalidLogin if basic auth credentials are provided).  However, that would break the existing behaviour, where if guest is enabled AND DisablePasswordAuth is true, basic auth credentials are ignored and it falls through to GUEST.  I'm not convinced this is the desired behaviour, but it's not a security issue and we don't want to break existing handling.